app/controllers/cart_products_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CartProductsController < ApplicationController
+  def create
+    products_in_cart = current_user.cart.products
+
+    if products_in_cart.map(&:id).include?(params[:product_id].to_i)
+      product_in_cart = products_in_cart.where(id: params[:product_id]).first
+
+      cart_product = CartProduct.find_by(product_id: product_in_cart, cart_id: current_user.cart.id)
+      cart_product.update(quantity: cart_product.quantity + 1)
+      cart_product.save
+    else
```

Comment:
If the product does not already exist in the cart then we just want to create a new `CartProduct` record.